### PR TITLE
Allow support for vllm batch with checkpoints

### PR DESCRIPTION
**Pull request by dmchoiboi**
# Pull Request Summary

_What is this PR changing? Why is this change being made? Any caveats you'd like to highlight? Link any relevant documents, links, or screenshots here if applicable._

Remove the requirement to add models to model zoo in order to run batch inference via vllm as long as a checkpoint is provided

## Test Plan and Usage Guide

_How did you validate that your PR works correctly? How do you run or demo the code? Provide enough detail so a reviewer can reasonably reproduce the testing procedure. Paste example command line invocations if applicable._

### CODE REVIEW

File: model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py [hunk 1]
```diff
@@ -370,10 +370,11 @@ def validate_checkpoint_path_uri(checkpoint_path: str) -> None:
 
 def get_checkpoint_path(model_name: str, checkpoint_path_override: Optional[str]) -> str:
     checkpoint_path = None
-    if SUPPORTED_MODELS_INFO[model_name].s3_repo:
-        checkpoint_path = get_models_s3_uri(SUPPORTED_MODELS_INFO[model_name].s3_repo, "")
+    models_info = SUPPORTED_MODELS_INFO.get(model_name, None)
```
Comment (edgan8) [inline model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py:373]: This LGTM but I'm worried there will be other errors if the model didn't have a preconfigured hardware profile. Looking forward to see what happens during testing!

Comment (dmchoiboi) [reply model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py:373]: hardware is determined based off hugging face repo config so there's no need to preconfigure hardware profile other than special cases

Comment (edgan8) [reply model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py:373]: OK SG, will approve after seeing what the test plan and test results look like

**Review by seanshi-scale — APPROVED** [review]

File: model-engine/model_engine_server/common/service_requests.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ def make_sync_request_with_retries(
             wait=wait_exponential(multiplier=1, min=1, max=timeout_seconds),
         ):
             with attempt:
-                logger.info(f"Retry number {attempt.retry_state.attempt_number}")
+                logger.debug(f"Retry number {attempt.retry_state.attempt_number}")
```
Comment (edgan8) [inline model-engine/model_engine_server/common/service_requests.py:40]: This is great will reduce logspam

**Pull request merged**